sdk/core/Azure.Core/src/Shared/NextLinkOperationImplementation.cs

```diff
@@ -70,6 +71,40 @@ internal class NextLinkOperationImplementation : IOperation
             return new OperationToOperationOfT<T>(operationSource, operation);
         }
 
+        public static IOperation Create(
+            HttpPipeline pipeline,
+            RehydrationToken? rehydrationToken,
+            string? apiVersionOverride = null)
+        {
+            Argument.AssertNotNull(rehydrationToken, nameof(rehydrationToken));
+
+            var lroDetails = ModelReaderWriter.Write(rehydrationToken!, new ModelReaderWriterOptions("J")).ToObjectFromJson<Dictionary<string, string>>();
+            if (!Uri.TryCreate(lroDetails["initialUri"], UriKind.Absolute, out var startRequestUri))
+            {
+                throw new InvalidOperationException("Invalid initial URI");
+            }
+
+            if (!lroDetails.TryGetValue("nextRequestUri", out var nextRequestUri))
+            {
+                throw new InvalidOperationException("Invalid next request URI");
+            }
+
+            RequestMethod requestMethod = new RequestMethod(lroDetails["requestMethod"]);
+            string lastKnownLocation = lroDetails["lastKnownLocation"];
+            OperationFinalStateVia finalStateVia = Enum.IsDefined(typeof(OperationFinalStateVia), lroDetails["finalStateVia"]) ? (OperationFinalStateVia)Enum.Parse(typeof(OperationFinalStateVia), lroDetails["finalStateVia"]) : OperationFinalStateVia.Location;
```

Comment:
It might be helpful for future maintainers to break up these statements a bit for readability.

Reply:
updated